src/acl/acl.internal.ts

```diff
@@ -491,3 +496,19 @@ export function internal_setAcl<ResourceExt extends WithServerResourceInfo>(
 ): ResourceExt & WithAcl {
   return Object.assign(resource, { internal_acl: acl });
 }
+
+export function internal_setResourceAcl<
+  T extends WithServerResourceInfo & WithAcl
+>(resource: T, acl: AclDataset): T & WithResourceAcl {
+  const newAcl: WithResourceAcl["internal_acl"] = {
+    resourceAcl: acl,
+    fallbackAcl: null,
```

Comment:
Just wondering if maybe we should check whether a fallback ACL is available for `resource` and, if so, preserve that?

Reply:
That was my initial though too, but the type definition actually prevents it: 
```
export type WithAcl = {
  internal_acl:
    | {
        resourceAcl: AclDataset;
        fallbackAcl: null;
      }
    | {
        resourceAcl: null;
        fallbackAcl: AclDataset | null;
      };
};
```

Which makes sense: as soon as a Resource has a Resource ACL, its Fallback ACL becomes irrelevant, so I guess overriding it isn't too much of an issue. Otherwise, we can update the type definition to allow both to coexist.

Reply:
Oh that's right, we changed it to only fetch the fallback ACL if the Resource ACL was not available, as indeed it's not relevant if there is a Resource ACL. Never mind then!